### PR TITLE
[Compaction] Modify compaction logic

### LINKED ISSUE #5715: Difficult to find Issue Tracker
It's not at all easy to find how to report bugs.

The only link I could find required navigating several pages.
The link on the page http://doris.incubator.apache.org/master/en/community/how-to-contribute.html#if-a-bug-or-problem-is-found leads to  https://github.com/apache/incubator-doris/issues/new/select which does not actually work.

I would expect to find a direct link to the Issue tracker in one of the menus as well.

### LINKED ISSUE #5738: [Compaction] A tablet cannot execute base and cumulative compaction at the same time
**Describe the bug**
As mentioned above, in a high-concurrency import scenario, if a tablet is doing base compaction and cannot perform cumulative compaction at the same time, and base compaction may run for a long time, this will cause the tablet to accumulate too many delta versions.

We can strictly guarantee that the cumulative point will not fall back, so it is safe for a tablet to do two compactions at the same time

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

1. Add /api/compaction/run_status to show the running compaction tasks.
2. Support do base and cumulative compaction for one tablet at same time.
3. Modify some log level.
4. Add a feedback document. [Close #5715 ]

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I have created an issue on (Fix #5738) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -283,8 +281,7 @@ CONF_mInt32(cumulative_compaction_skip_window_seconds, "30");
 CONF_mInt64(min_compaction_failure_interval_sec, "600"); // 10 min
 
 // This config can be set to limit thread number in compaction thread pool.
-CONF_mInt32(min_compaction_threads, "10");
-CONF_mInt32(max_compaction_threads, "10");
+CONF_Int32(max_compaction_threads, "10");
```
Comment (weizuo93) [inline be/src/common/config.h:284]: Now, `max_threads` and `min_threads`of Thread Pool can not be adjusted dynamically. Recently, I am looking into the implementation of the ThreadPool, and I consider supporting dynamic adjustment of `max_threads` and `min_threads`for thread pool later. How do you like it?

File: be/src/http/action/compaction_action.cpp [hunk 2]
```diff
@@ -155,59 +154,66 @@ Status CompactionAction::_handle_run_status_compaction(HttpRequest* req, std::st
...
-        MutexLock lock_cumulative(tablet->get_cumulative_lock(), TRY_LOCK);
-        if (!lock_cumulative.own_lock()) {
-            msg = "this tablet_id is running";
-            compaction_type = "cumulative";
-            run_status = 1;
-            *json_result = strings::Substitute(json_template, run_status, msg, tablet_id,
-                                               schema_hash, compaction_type);
-            return Status::OK();
+    
+        std::string msg = "this tablet_id is not running";
```
Comment (weizuo93) [inline be/src/http/action/compaction_action.cpp:183]: I think `compaction task for this tablet_id is not running` may be better, and `this tablet_id is not running`may be misunderstood to mean that this tablet is unhealthy.

File: be/src/http/action/compaction_action.cpp [hunk 3]
```diff
@@ -155,59 +154,66 @@ Status CompactionAction::_handle_run_status_compaction(HttpRequest* req, std::st
...
+        std::string compaction_type = "";
+        bool run_status = 0;
+    
+        {
+            // use try lock to check this tablet is running cumulative compaction
+            MutexLock lock_cumulative(tablet->get_cumulative_lock(), TRY_LOCK);
+            if (!lock_cumulative.own_lock()) {
+                msg = "this tablet_id is running";
```
Comment (weizuo93) [inline be/src/http/action/compaction_action.cpp:191]: `this tablet_id is running compaction task`

File: be/src/http/action/compaction_action.cpp [hunk 4]
```diff
@@ -155,59 +154,66 @@ Status CompactionAction::_handle_run_status_compaction(HttpRequest* req, std::st
...
-            run_status = 1;
-            *json_result = strings::Substitute(json_template, run_status, msg, tablet_id,
-                                               schema_hash, compaction_type);
-            return Status::OK();
+    
+        {
+            // use try lock to check this tablet is running base compaction
+            MutexLock lock_base(tablet->get_base_lock(), TRY_LOCK);
+            if (!lock_base.own_lock()) {
+                msg = "this tablet_id is running";
```
Comment (weizuo93) [inline be/src/http/action/compaction_action.cpp:204]: `this tablet_id is running compaction task`?

**Comment by weizuo93** [comment]
+1

File: be/src/common/config.h (hunk 1 above)
Comment (morningman) [reply be/src/common/config.h:284]: It's OK for me~
But I think the `min_compaction_threads` is meaning less for user, we should reduce unnecessary configuration items as much as possible

File: be/src/olap/storage_engine.cpp [hunk 5]
```diff
@@ -1035,4 +1035,76 @@ void StorageEngine::create_base_compaction(TabletSharedPtr best_tablet,
     base_compaction.reset(new BaseCompaction(best_tablet, tracker_label, _compaction_mem_tracker));
 }
 
+// Return json:
+// {
+//   "CumulativeCompaction": {
+//          "/home/disk1" : [10001, 10002],
```
Comment (weizuo93) [inline be/src/olap/storage_engine.cpp:1041]: Can we add segment number that be merged in this compaction task for each thread? Of course, we could add the segment number info in another PR.

File: be/src/http/action/compaction_action.cpp (hunk 2 above)
Comment (morningman) [reply be/src/http/action/compaction_action.cpp:183]: ok

File: be/src/common/config.h (hunk 1 above)
Comment (weizuo93) [reply be/src/common/config.h:284]: > It's OK for me~
> But I think the `min_compaction_threads` is meaning less for user, we should reduce unnecessary configuration items as much as possible

OK.

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**